### PR TITLE
NEWS: Updated NEWS for 1.15.0-rc1 and 1.15.0-rc2.

### PR DESCRIPTION
**Pull request by rakhmets**
## What
Updated `NEWS` file filling 1.15.0-rc1 and 1.15.0-rc2. sections.
Added 1.14.1 section.

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -11,9 +11,171 @@
...
+
+#### Documentation
+* Updated ucp_worker_release_address description
+
+#### Tests
+* Disabled gtests on wireless interfaces
+* Disabled JAVA tests on docker interfaces
+* Fixed wrong usage of ep_close in examples
+* Fixed failures on tests built in single-treaded mode
+* Multiple fixes in unit tests
```
Comment (brminich) [inline NEWS:163]: maybe mention just `Fixed wrong usage of ep_close in examples`, as all the rest seems to be infra internals

**Review by brminich — COMMENTED** [review]
i think we also need to port NEWS for v1.14.1 here

File: NEWS [hunk 2]
```diff
@@ -11,9 +11,171 @@
 ### Features:
 ### Bugfixes:
 
+## 1.15.0-rc2 (July 27, 2023)
+
+### Features:
+
+#### RDMA CORE (IB, ROCE, etc.)
+* Implemented is_reachable_v2 for IB interfaces
+
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply NEWS]: Removed.

File: NEWS [hunk 3]
```diff
@@ -11,9 +11,171 @@
...
+* Fixed performance degradation when KSM with atomic access fails repeatedly during rkey_pack phase of rendezvous protocol
+* Fixed invalid access error to strict-order key with offset
+
+#### GPU (CUDA, ROCM)
+* Fixed locking of host pinned memory on rocm
+* Fixed CUDA IPC performance degradation after libnuma removal
+
+#### Tests
+* Temporary disabled job failures on io_demo counter
+* Fixed unit test failures
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply NEWS]: Removed.

File: NEWS [hunk 4]
```diff
@@ -11,9 +11,171 @@
...
+* Added a rocm docker container for testing
+* Added option to send client Id in IODEMO test
+* Added support for repeating connection to the same server in IODEMO test
+* Added receive callback for receiving FIN message in UCX examples
+
+#### Tools
+* Added user-side memcpy option for AM benchmarks in ucx_perftest
+
+#### Build
+* Added UCX LUA dissectors for 64-bit little-endian
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply NEWS:83]: Added

File: NEWS (hunk 1 above)
Comment (rakhmets) [reply NEWS]: Updated.

**Comment by rakhmets** [comment]
> i think we also need to port NEWS for v1.14.1 here

Wouldn't that be ambiguous? It can be interpreted as all the listed changes are included to the release...

**Comment by rakhmets** [comment]
> > i think we also need to port NEWS for v1.14.1 here
> 
> Wouldn't that be ambiguous? It can be interpreted as all the listed changes are included to the release...

It seems that everything is included. I will add NEWS for v1.14.1, and remove duplications.

**Comment by brminich** [comment]
> 

I meant add a separate section for v1.14.1, which is missing now (was not merged to this branch when 1.14.1 was created)

File: NEWS [hunk 5]
```diff
@@ -11,9 +11,143 @@
...
+#### Tests
+* Fixed wrong usage of ep_close in examples
+#### Tools
+* Removed support for librte from perf
+* Fixed worker flush deadlock when using multiple workers in ucx_perftest
+#### Build
+* Changed 'unsupported option' ICC command line warning to error
+* Removed never used fault-injection configuration option
+* Fixed obsolete macro warnings in new autoconf/libtool
+* Fixed building UCX with non-old GCC
```
Comment (brminich) [inline NEWS:132]: maybe `with modern GCC versions`?

File: NEWS [hunk 6]
```diff
@@ -11,9 +11,143 @@
...
+* Added a rocm docker container for testing
+* Added option to send client Id in IODEMO test
+* Added support for repeating connection to the same server in IODEMO test
+* Added receive callback for receiving FIN message in UCX examples
+#### Tools
+* Added user-side memcpy option for AM benchmarks in ucx_perftest
+* Added wireshark LUA dissectors for some UCX protocols
+#### Build
+* Added a separate xpmem deb subpackage
+* Added ARM-based artifacts.
```
Comment (brminich) [inline NEWS:81]: please remove the `.` in the end (for consistency)

Comment (rakhmets) [reply NEWS:81]: Removed

File: NEWS [hunk 7]
```diff
@@ -11,9 +11,143 @@
...
+### Features:
+#### RDMA CORE (IB, ROCE, etc.)
+* Implemented is_reachable_v2 for IB interfaces
+#### Build
+* Enabled build with binutils 2.40
+* Removed inbox packages on install
+
+### Bugfixes:
+#### UCP
+* Reverted the change in lanes selection to fix broken wire compatibility
```
Comment (shamisp) [inline NEWS:24]: up to version ?

File: NEWS [hunk 8]
```diff
@@ -11,9 +11,143 @@
...
+* Fixed endpoint reconfiguration error due to wrong endpoint configuration locality assignment
+#### RDMA CORE (IB, ROCE, etc.)
+* Fixed performance degradation when KSM with atomic access fails repeatedly during rkey_pack phase of rendezvous protocol
```
Comment (shamisp) [inline NEWS:27]: too much details. Nobody knows what is KSM

**Comment by brminich** [comment]
> The patch will conflict with another patch back porting news from 1.14.1

No, #9247 is targeting master, while this one updates v1.15.x branch

File: NEWS (hunk 8 above)
Comment (rakhmets) [reply NEWS]: Removed details.

File: NEWS (hunk 7 above)
Comment (rakhmets) [reply NEWS]: it was broken in 1.15.0-rc1.

File: NEWS [hunk 9]
```diff
@@ -11,9 +11,143 @@
...
+* Added wireshark LUA dissectors for some UCX protocols
+#### Build
+* Added a separate xpmem deb subpackage
+* Added ARM-based artifacts
+* Removed dependency on libnuma
 
+### Bugfixes:
+#### UCP
+* Fixed crash during connection manager cleanup
+* Fixed AM send status handler in the new protocol infrastructure
```
Comment (yosefe) [inline NEWS:87]: what is this? maybe not so important?

File: NEWS [hunk 10]
```diff
@@ -11,9 +11,143 @@
...
+* Fixed rkey index calculation for Rendezvous protocol
+* Fixed rcache dump function
+* Removed logging from rkey unpack in release mode
+* Removed double resource cleanup from Rendezvous rkey protocol
```
Comment (yosefe) [inline NEWS:91]: Fixed dobule free of rkey in rendezvous protocol

File: NEWS [hunk 11]
```diff
@@ -11,9 +11,143 @@
...
+* Fixed rtr/ppl error flow
```
Comment (yosefe) [inline NEWS:92]: Fixed rendezvous pipeline protocol error flow

File: NEWS [hunk 12]
```diff
@@ -11,9 +11,143 @@
...
+* Added workaround to error handling of rndv/get/zcopy
```
Comment (yosefe) [inline NEWS:93]: Fixed error handling in rendezvous get zcopy protocol

File: NEWS [hunk 13]
```diff
@@ -11,9 +11,143 @@
...
+* Replay pending requests of wireup EP CM during connection establishment to prevent potential ordering issues and wrong configuration
+* Pass user-provided memory type to the function that checks whether the buffer can be sent inline or not
+* Removed test memory registration during UCP context initialization
```
Comment (yosefe) [inline NEWS:96]: Avoid memory registration during UCP context initialization

File: NEWS [hunk 14]
```diff
@@ -11,9 +11,143 @@
...
+* Fixed CPU/device atomics selection in the new protocol infrastructure
+* Multiple fixes in the new protocol infrastructure information output
+#### UCT
+* Fixed exported memh packing
+* Fixed an error in checking return status of multi-threaded memory registration function
+* Introduced MD capability for non-blocking registration memory types
```
Comment (yosefe) [inline NEWS:102]: isn't it a new feature?

File: NEWS [hunk 15]
```diff
@@ -11,9 +11,143 @@
...
+#### RDMA CORE (IB, ROCE, etc.)
+* Added check for UAR support to memory domain opening
```
Comment (yosefe) [inline NEWS:104]: UAR or UMR?

Comment (rakhmets) [reply NEWS:101]: UAR - User Access Region

File: NEWS [hunk 16]
```diff
@@ -11,9 +11,143 @@
...
+#### RDMA CORE (IB, ROCE, etc.)
+* Implemented is_reachable_v2 for IB interfaces
+#### Build
+* Enabled build with binutils 2.40
+* Removed inbox packages on install
+
+### Bugfixes:
+#### UCP
+* Reverted the change in lanes selection to fix broken wire compatibility
+* Fixed endpoint reconfiguration error due to wrong endpoint configuration locality assignment
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply NEWS]: fixed

File: NEWS [hunk 17]
```diff
@@ -11,9 +11,143 @@
...
+#### RDMA CORE (IB, ROCE, etc.)
+* Fixed performance degradation when KSM with atomic access fails repeatedly during rkey_pack phase of rendezvous protocol
+* Fixed invalid access error to strict-order key with offset
+#### GPU (CUDA, ROCM)
+* Fixed CUDA IPC performance degradation after libnuma removal
+#### Build
+* Updated tags for CUDA 11 images, as older versions were removed from Dockerhub
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply NEWS]: removed

File: NEWS [hunk 18]
```diff
@@ -11,9 +11,143 @@
...
+* Updated allow/deny transport list feature to control auxiliary transport selection
+* Multiple performance improvements of the new protocol infrastructure
+* Multiple improvements in error and debug messages
+#### UCT
+* Split UCT_MD_MKEY_PACK_FLAG_INVALIDATE into two flags for RMA and AMO
+* Added put_zcopy and get_zcopy scheme support for self transport
+* Added base implementation of is_reachable_v2 API using intra/inter flag
+#### RDMA CORE (IB, ROCE, etc.)
+* Added option to control CQE zipping per CQ RX/TX direction
+* Added option to specify how DCI select port under RoCE LAG
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply NEWS]: fixed

File: NEWS [hunk 19]
```diff
@@ -11,9 +11,143 @@
...
+* Improved performance estimation of cuda_ipc on Hopper architecture
+* Added rcache parameters for rocm transports
+* Introduced dmabuf support for rocm transports
+* Implemented asynchronous progress for the zcopy operations in the rocm_copy transport
+* Added option to enable using cross-device dmabuf file descriptor for rocm
+#### Java
+* Added Java bindings for exported memh feature
+#### Tests
+* Added a rocm docker container for testing
+* Added option to send client Id in IODEMO test
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply NEWS]: updated

File: NEWS [hunk 20]
```diff
@@ -11,9 +11,143 @@
...
+* Added support for repeating connection to the same server in IODEMO test
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply NEWS]: updated

File: NEWS [hunk 21]
```diff
@@ -11,9 +11,143 @@
...
+* Added receive callback for receiving FIN message in UCX examples
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply NEWS]: updated

File: NEWS [hunk 22]
```diff
@@ -11,9 +11,143 @@
...
+#### Tools
+* Added user-side memcpy option for AM benchmarks in ucx_perftest
+* Added wireshark LUA dissectors for some UCX protocols
+#### Build
+* Added a separate xpmem deb subpackage
+* Added ARM-based artifacts
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply NEWS]: updated

File: NEWS (hunk 9 above)
Comment (rakhmets) [reply NEWS]: removed

File: NEWS [hunk 23]
```diff
@@ -11,9 +11,143 @@
...
+* Removed dependency on libnuma
 
+### Bugfixes:
+#### UCP
+* Fixed crash during connection manager cleanup
+* Fixed AM send status handler in the new protocol infrastructure
+* Fixed rkey index calculation for Rendezvous protocol
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply NEWS]: fixed

File: NEWS (hunk 11 above)
Comment (rakhmets) [reply NEWS]: fixed

File: NEWS (hunk 12 above)
Comment (rakhmets) [reply NEWS]: fixed

File: NEWS (hunk 13 above)
Comment (rakhmets) [reply NEWS]: fixed

File: NEWS (hunk 14 above)
Comment (rakhmets) [reply NEWS]: moved to features

File: NEWS [hunk 24]
```diff
@@ -11,9 +11,143 @@
 ### Features:
 ### Bugfixes:
 
+## 1.15.0-rc2 (July 27, 2023)
+### Features:
+#### RDMA CORE (IB, ROCE, etc.)
+* Implemented is_reachable_v2 for IB interfaces
+#### Build
+* Enabled build with binutils 2.40
+* Removed inbox packages on install
```
[the remark this answers is not in the document]
Comment (rakhmets) [reply NEWS]: Changed to: Added versioned dependency to switch between packages with the same names

File: NEWS [hunk 25]
```diff
@@ -11,9 +11,140 @@
...
+### Features:
+#### RDMA CORE (IB, ROCE, etc.)
+* Implemented is_reachable_v2 for IB interfaces
+#### Build
+* Enabled build with binutils 2.40
+* Added versioned dependency to switch between packages with the same names
+
+### Bugfixes:
+#### UCP
+* Reverted the change in lanes selection to fix wire compatibility broken in 1.15.0-rc1
```
Comment (rakhmets) [inline NEWS:24]: https://github.com/openucx/ucx/pull/9244#discussion_r1275874176
@yosefe the compatibility was broken in rc1, and fixed in rc2. Thus, it shouldn't be mentioned in final news for 1.15.0.
Is the line is ok now?

Comment (yosefe) [reply NEWS:24]: Better remove this line IMO

**Review by yosefe — DISMISSED** [review]

File: NEWS (hunk 25 above)
Comment (rakhmets) [reply NEWS]: Removed

**Review by yosefe — APPROVED** [review]

**Comment by shamisp** [comment]
Looks good ! Great progress !

**Review by shamisp — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Pull request merged**